game/lib/widget/leaderboard_cell.dart

```diff
@@ -34,6 +34,13 @@ Widget leaderBoardCell(
     height: 0,
   );
 
+// Helper Function with ellipses functionality such that we can set max character length length/cutoff
+  String truncateWithEllipsis(String text, int maxLength) {
+    return (text.length <= maxLength)
+        ? text
+        : '${text.substring(0, maxLength)}...';
```

Comment:
This looks great! However, it may be worthwhile to consider the event where the text has emojis or other special characters. Since these can take up more than one character, truncating them could cause the character to break and produce an odd string of text.